### PR TITLE
feat.: Add nda_caselist_filter.py to select subjects based on the following criteria

### PR DESCRIPTION
**Pull request by dheshanm**
Subjects are selected based on the following criteria:

image

If a subject meets the criteria, they are retained. Otherwise, they are skipped, and the reason for them being skipped is noted.

### CODE REVIEW

File: nda_caselist_filter.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+logger = logging.getLogger(__name__)
+logargs = {
+    "level": logging.DEBUG,
+    # "format": "%(asctime)s - %(process)d - %(name)s - %(levelname)s - %(message)s",
+    "format": "%(asctime)s - %(process)d - %(name)s - %(filename)s:%(lineno)d - %(levelname)s - "
+    "%(message)s",
+}
+logging.basicConfig(**logargs)
+
+DEST_CSV_PATH = "/data/predict1/home/dm1447/data/nda/ndar_subject01_selected_v2.csv"
```
Comment (tashrifbillah) [inline nda_caselist_filter.py:34]: Make it `sys.argv[1]`.

File: nda_caselist_filter.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    ndar_csvs = list(
+        Path("/data/predict1/to_nda/nda-submissions").glob("ndar_subject01*.csv")
+    )
+    logger.info(f"Reading ndar_subject01 files: {ndar_csvs}")
+
+    ndar_df = pd.DataFrame()
+
+    for ndar_csv in ndar_csvs:
+        # Reak skipping the first row
+        temp_df = pd.read_csv(ndar_csv, skiprows=1)
```
Comment (tashrifbillah) [inline nda_caselist_filter.py:55]: Use `dtyp=str` argument.

File: nda_caselist_filter.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+from typing import Any, Dict, List, Optional, Tuple
+
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+logargs = {
+    "level": logging.DEBUG,
+    # "format": "%(asctime)s - %(process)d - %(name)s - %(levelname)s - %(message)s",
+    "format": "%(asctime)s - %(process)d - %(name)s - %(filename)s:%(lineno)d - %(levelname)s - "
+    "%(message)s",
```
Comment (tashrifbillah) [inline nda_caselist_filter.py:30]: Wasn't the commented line a better (concise) log?

Comment (dheshanm) [reply nda_caselist_filter.py:30]: The added `%(filename)s:%(lineno)d` information was helpful during debugging. I can switch over to the more concise logs!

File: nda_caselist_filter.py (hunk 2 above)
Comment (dheshanm) [reply nda_caselist_filter.py]: Addressed this in recent commit!

**Comment by tashrifbillah** [comment]
Could you comment which rows from the criteria you implemented or used your database, and which rows make use of Owen's work?

**Comment by tashrifbillah** [comment]
In future, we may be interested to make it standalone i.e. implement all within script or use software engineer's database. Because as Nora pointed out the other day--Owen's work was not reliable.

**Comment by dheshanm** [comment]
> Could you comment which rows from the criteria you implemented or used your database, and which rows make use of Owen's work?

Only criteria number 2 is checked from directly from the JSONs. (I removed using internal DB in favor of using local JSONs, so this script is already standalone)

Other criteria: 3, 4, 5 and 6 are derived from Owen's QC Tracker. I check his `qc_tracker` for all clear and skip them if they have any issues flagged by Owen, (I used to check 4 and 5 myself, but later Owen had added more comprehensive checks, and not I use his checks instead).

**Pull request merged**